Detectors/CPV/calib/include/CPVCalibration/NoiseCalibrator.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef CPV_NOISE_CALIBRATOR_H_
+#define CPV_NOISE_CALIBRATOR_H_
+
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "DataFormatsCPV/Digit.h"
+#include "DataFormatsCPV/BadChannelMap.h"
+#include "CCDB/CcdbObjectInfo.h"
+
+namespace o2
+{
+namespace cpv
+{
+//=========================================================================
+using Digit = o2::cpv::Digit;
+struct NoiseCalibData {
+  int mNEvents = 0;
+  int mNoiseThreshold = 10; //ADC counts threshold
+  std::vector<int> mOccupancyMap;
+
+  NoiseCalibData();
+  ~NoiseCalibData() = default;
+
+  void fill(const gsl::span<const o2::cpv::Digit> data);
+  void merge(const NoiseCalibData* prev);
+  void print();
+
+}; // end NoiseCalibData
+//=========================================================================
+
+using NoiseTimeSlot = o2::calibration::TimeSlot<o2::cpv::NoiseCalibData>;
+class NoiseCalibrator final : public o2::calibration::TimeSlotCalibration<o2::cpv::Digit, o2::cpv::NoiseCalibData>
+{
+ public:
+  NoiseCalibrator();
+  ~NoiseCalibrator() final = default;
+  std::vector<o2::ccdb::CcdbObjectInfo> getCcdbInfoBadChannelMapVector() { return mCcdbInfoBadChannelMapVec; }
+  std::vector<o2::cpv::BadChannelMap> getBadChannelMapVector() { return mBadChannelMapVec; }
+
+  bool hasEnoughData(const NoiseTimeSlot& slot) const final
+  {
+    LOG(info) << "hasEnoughData() is being called";
+    return slot.getContainer()->mNEvents >= mMinEvents;
+  }
+  void initOutput() final;
+  void finalizeSlot(NoiseTimeSlot& slot) final;
+  NoiseTimeSlot& emplaceNewSlot(bool front, uint64_t tstart, uint64_t tend) final;
+
+  std::vector<float>* mPedEfficiencies = nullptr;
+  std::vector<int>* mDeadChannels = nullptr;
+  std::vector<int>* mHighPedChannels = nullptr;
```

Comment:
if you are using a class (rather than a struct) then move the datamembers to `protected:` part and have getters/setters. Also, you are assigning a to these data members arrays created with `new`, but then you never call `delete[]`. The best would be to use simply vectors (instead of pointers) or use e.g. `std::unique_ptr<std::vector<float>> mPedEfficiencies`, which you can set by external raw vector pointer by `mPedEfficiencies.reset(ptr)`.